tests/integration/conditions/test_ecdsa_condition_integration.py

```diff
@@ -129,30 +144,257 @@ def test_ecdsa_in_compound_condition():
     assert and_lingo.eval(**context) is True
 
 
-def test_discord_ed25519_with_ecdsa_condition():
-    # Discord Ed25519 test vector
-    public_key_hex = "b853dd9f496723daf64bf2f5a886086f790df66e61d7b6f7f98a50c9e5ede8f3"
-    signature_hex = "0a12acb96843700b724f1c9dba3075a7fc482677e0c713eb6be63bfea406fb33b4715975f1062b2dff95127bd559ee23758911bd217760727dc44e7880bc6e04"
-    timestamp = "1749368683"
-    body = '{"app_permissions":"2248473465835073","application_id":"1380486651436073092","attachment_size_limit":10485760,"authorizing_integration_owners":{"0":"1380488052169769110"},"channel":{"flags":0,"guild_id":"1380488052169769110","icon_emoji":{"id":null,"name":"👋"},"id":"1380488052169769113","last_message_id":"1380528253168652328","name":"general","nsfw":false,"parent_id":"1380488052169769111","permissions":"2251799813685247","position":0,"rate_limit_per_user":0,"theme_color":null,"topic":null,"type":0},"channel_id":"1380488052169769113","context":0,"data":{"id":"1380515955146358918","name":"sign","options":[{"name":"message","type":3,"value":"llamas"}],"type":1},"entitlement_sku_ids":[],"entitlements":[],"guild":{"features":[],"id":"1380488052169769110","locale":"en-US"},"guild_id":"1380488052169769110","guild_locale":"en-US","id":"1381177107555680327","locale":"en-US","member":{"avatar":null,"banner":null,"communication_disabled_until":null,"deaf":false,"flags":0,"joined_at":"2025-06-06T10:06:39.888000+00:00","mute":false,"nick":null,"pending":false,"permissions":"2251799813685247","premium_since":null,"roles":[],"unusual_dm_activity_until":null,"user":{"avatar":"9c5483a989a10edc8b831b6c8f284724","avatar_decoration_data":null,"clan":null,"collectibles":null,"discriminator":"0","global_name":"kprasch","id":"410212090289192960","primary_guild":null,"public_flags":0,"username":"kprasch"}},"token":"aW50ZXJhY3Rpb246MTM4MTE3NzEwNzU1NTY4MDMyNzo2ZlVlTkdIQVFIdkdhNUN2bXlYZ1RCRkZ5YkpuVm9WUG5Bbjl0TTMyUkFTenZGYXNXclBCMjZWUWJlalczcllRak9sc0JZR3Q4WW9sODBRcDg2c2hrYmRkYzlWcjF3TjdlRFVDMVBVTkZ3Z2VmRUp4VkI1MkZGNmVvM3hkWXd1Qg","type":2,"version":1}'
+def test_discord_json_from_body_string():
+    """Test parsing Discord body JSON string directly from context variable with 3-variable context"""
+    # Use Discord Ed25519 test vectors
+    public_key_hex = DISCORD_PUBLIC_KEY_HEX
+    signature_hex = DISCORD_SIGNATURE_HEX
+    timestamp = DISCORD_TIMESTAMP
 
-    # Construct the message to be signed
-    message = timestamp.encode("utf-8") + body.encode("utf-8")
+    # This is the raw JSON string as it would come from Discord
+    body = DISCORD_BODY
 
-    # Create ECDSACondition using the Discord public key and signature
+    # Create ECDSA condition for signature verification with message template
     ecdsa_condition = ECDSACondition(
-        message=":message",
+        message=":timestamp:discordPayload",  # Constructs message from timestamp + payload
```

Comment:
I accidentally left this in... I’ll refactor to remove it entirely (unless we decide this type of expression is useful?).